sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -427,6 +430,44 @@ private[hive] class HiveClientImpl(
     client.alterTable(qualifiedTableName, hiveTable)
   }
 
+  override def alterTableStats(table: CatalogTable): Unit = withHiveState {
```

Comment:
Based on this, we should move alterTableStats to `HiveExternalCatalog`, and still keep constructStatsFromHive in `HiveClientImpl`, right?